packit_service/worker/reporting.py

```diff
@@ -240,7 +241,37 @@ def report_status_by_comment(
     def get_statuses(self):
         self.project_with_commit.get_commit_statuses(commit=self.commit_sha)
 
-    def comment(self, body: str):
+    def _last_comment_identical(self, body: str) -> bool:
+        """Checks if the body is the same as the last comment."""
+        if self.pr_id:
+            for comment in self.project.get_pr(pr_id=self.pr_id).get_comments(
+                reverse=True
+            ):
+                if comment.author.startswith(PACKIT_APP_PREFIX):
+                    return body == comment.body
+        else:
+            commit_comments = self.project.get_commit_comments(self.commit_sha)
+            for comment in reversed(commit_comments):
+                if comment.author.startswith(PACKIT_APP_PREFIX):
+                    return body == comment.comment
+
+        return False
+
+    def comment(self, body: str, check_for_duplicate: bool = False):
+        """Reports status by adding a comment.
+
+        If the instance is tied to a PR, the comment is added to the PR,
+        otherwise, it is added to the tied commit (if the forge supports
+        commit comments).
+
+        Args:
+            body: The comment text.
+            check_for_duplicate: If set to True, the comment won't be added if
+                it is already present in the PR/commit as the last comment.
```

Comment:
nice refactoring! I wonder whether we want to check only the last comment or avoid duplicity at all or have options to do either, WDYT? E.g. for the copr restriction messages, it could be enough to comment once per event.

Reply:
that sounds perfect